packages/lodestar/src/network/gossip/gossip.ts

```diff
@@ -123,15 +128,59 @@ export class Gossip extends (EventEmitter as { new(): GossipEventEmitter }) impl
     subnet: number|string,
     callback?: (attestation: {attestation: Attestation; subnet: number}) => void
   ): void {
-    this.subscribe(forkDigest, GossipEvent.ATTESTATION_SUBNET, callback, new Map([["subnet", subnet.toString()]]));
```

Comment:
Can't use the common `subscribe()` method here because 1 GossipEvent.ATTESTATION_SUBNET maps to multiple gossip topics.
Use the same strategy with new `attNetListenerCount`

Reply:
I think common subscribe method works fine here.
generic subscribe method create topic from GossipEvent.ATTESTATION_SUBNET, forkDigest and subnet so it should work as expected. we have unique topic for each subnet

Reply:
yeah the `subscribe` is unique but they share the same `GossipEvent.ATTESTATION_SUBNET` event. Given this code `this.on(event, listener as (...args: unknown[]) => void);` in subscribe method, it'll call the callback function of all callbacks. And we can't base on `this.listenerCount(event)` to subscribe/unsubscribe to pubsub because it's common to all subnet so I have to manage the listener count for each topic in `this.attNetListenerCount`

Reply:
I see what you are saying now.
But still, you should change this code a bit to use existing EventEmmiteer, possible solution is 
`this.on(GossipEvent.ATTESTATION_SUBNET +"_" + subnet, listener)`

If you modify generic subscribe to accept custom string, you could pass `GossipEvent.ATTESTATION_SUBNET +"_" + subnet` instead of just `keyof IGossipEvents` so you can reuse it. Just make sure you change it gossip handler